go/kex2/transport.go

```diff
@@ -339,12 +370,17 @@ func (c *Conn) pollLoop(poll time.Duration) (msgs [][]byte, err error) {
 
 	var totalWaitTime time.Duration
 
+	c.setPollLoopRunning(true)
+	defer func() {
+		c.setPollLoopRunning(false)
```

Comment:
this is for testing, but i abandoned the tests since I couldn't get them to work.  when tests are put back in this will be useful.